src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -5068,6 +5068,7 @@ internal enum AddressKind
 
                 case BoundKind.Parameter:
                     return IsAnyReadOnly(addressKind) ||
+                        ((BoundParameter)expression).Type.TypeKind == TypeKind.Enum ||
```

Comment:
This only covers readonly references from parameters, there are many other places readonly references can come from. Probably better to make changes like so:

```diff
-internal static bool IsAnyReadOnly(AddressKind addressKind) => addressKind >= AddressKind.ReadOnly;
+internal static bool IsAnyReadOnly(AddressKind addressKind, bool isEnum) => addressKind >= AddressKind.ReadOnly || (isEnum && addressKind >= AddressKind.Constrained);
```
and update the callers to pass in whether the instance is an enum there.

Reply:
I am very confused by this suggestion. In the test I added, `addressKind` is writeable meaning if I were to do this change instead of what I did, it would still return false here.

But the bigger problem is I don't understand this change in general: is this saying an enum is readonly when the address kind is constrained? That seems off to me.